pkg/registry_checker/checker.go

```diff
@@ -193,7 +194,17 @@ func NewRegistryChecker(
 	}
 	rc.controllerIndexers.cronJobIndexer = rc.cronJobsInformer.Informer().GetIndexer()
 
-	rc.controllerIndexers.secretIndexer = rc.secretsInformer.Informer().GetIndexer()
+	namespace := "default"
+	// Create a context
+	ctx := context.TODO()
+	// Attempt to list secrets in the default namespace
+	_, enumerr := kubeClient.CoreV1().Secrets(namespace).List(ctx, metav1.ListOptions{})
+	if enumerr != nil {
+		// Not add the secret indexer to automatic cache updater
+		logrus.Warn("Provided ServiceAccount does not seem to be able to list secrets. Image availability check for images in private registries not having spec.imagePullSecrets configured will fail!")
```

Comment:
I would be more clear to the reader, maybe like that: 
```suggestion
		logrus.Warn("The provided ServiceAccount is not able to list secrets. The check for images in private registries requires 'spec.imagePullSecrets' to be configured correctly.")
```